salt/_modules/metalk8s_volumes.py

```diff
@@ -377,6 +437,79 @@ def is_cleaned_up(self):
     def clean_up(self):
         return  # Nothing to do
 
+class RawBlockDeviceNoFormat(RawBlockDevice):
+    def __init__(self, volume):
+        super(RawBlockDeviceNoFormat, self).__init__(volume)
+        # Detect which kind of device we have: a real disk, only a partition or
+        # an LVM volume.
+        name = device_name(self.path)
+        self._partition = None
+        for symlink in glob.glob('/dev/disk/by-id/dm-uuid-LVM-*'):
+            realpath = os.path.realpath(symlink)
+            if os.path.basename(realpath) == name:
+                self._kind = DeviceType.LVM
+                return
+        match = re.search('(?P<partition>\d+)$', name)
+        if match is not None:
+            self._kind = DeviceType.PARTITION
+            self._partition = match.groupdict()['partition']
+        else:
+            self._kind = DeviceType.DISK
+
+    @property
+    def persistent_path(self):
+        if self._kind == DeviceType.LVM:
+            name = device_name(self.path)
+            for symlink in glob.glob('/dev/disk/by-id/dm-uuid-LVM-*'):
+                realpath = os.path.realpath(symlink)
+                if os.path.basename(realpath) == name:
+                    return symlink
+        return '/dev/disk/by-partlabel/{}'.format(self.get('metadata.uid'))
+
+    @property
+    def device_path(self):
+        path = self.path
+        if self._kind == DeviceType.PARTITION:
+            path = path.rstrip(self._partition)
+        return path
+
+    @property
+    def is_prepared(self):
+        # Nothing to do in LVM case (we use LVM UUID).
+        if self._kind == DeviceType.LVM:
+            return True
+        partition = self._partition or '1'
+        pattern = '{}.+{}'.format(partition, self.get('metadata.uid').lower())
+        result  = _run_cmd(' '.join(['parted', self.device_path, 'print']))
+        return re.search(pattern, result['stdout']) is not None
+
+    def prepare(self, force=False):
+        # Nothing to do in LVM case.
+        if self._kind == DeviceType.LVM:
+            return
+        uuid = self.get('metadata.uid')
+        # In case of partition, set the label to volume UUID.
```

Comment:
Setting the *label* to the volume UUID? Can't we change the partition UUID instead?

Reply:
Yeah, I wanted to use partition UUID at first, but couldn't find a way to set it :confused:  so I went with partition label instead.

Reply:
Not fond of that at all, because super confusing. Instead, I'd set the *name* of the GPT partition to the `metadata.name` of the `Volume` object...

Reply:
I used UUID for consistency with the others, but I can change it to `name`.

Reply:
I'm not sure we're aligned here :)

In GPT, a partition can have a *name* and a *GUID*. So, what I suggest is to set its *GUID* to the `metadata.uid` of the `Volume` object, and the *name* to `metadata.name` of the `Volume`. Then, always use the `/dev/disk/by-partuuid/...` path to *refer* to it. The *name* is merely cosmetic.

Reply:
Yup, but:

> I wanted to use partition UUID at first, but couldn't find a way to set it :confused: 

Haven't find a way to set the partition UID with parted.
Will give a try to `sgdisk`, maybe it has more options.

Reply:
Easy to do with `sgdisk`, I've added a new commit to use `sgdisk` instead of `parted`.